src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php

```diff
@@ -25,6 +25,10 @@ class PhpSerializer implements SerializerInterface
      */
     public function decode(array $encodedEnvelope): Envelope
     {
+        if (!\array_key_exists('body', $encodedEnvelope)) {
+            throw new MessageDecodingFailedException(__CLASS__.' only supports envelope with "body" index. Maybe you should implement your own serializer?');
+        }
```

Comment:
Given that the `empty()` check below already covers the case where the key does not exist, could we just change the message of the exception thrown there instead of adding another one?